initial commit Diplom1

EgorKoshelev96 opened the pull request.

EgorKoshelev96 committed:
initial commit Diplom1

EgorKoshelev96 committed:
add jacoco report

EgorKoshelev96 committed:
Create README.md

EgorKoshelev96 commented:
Задание 1: юнит-тесты

EgorKoshelev96 committed:
-- Для тестов с несколькими проверками использовал softAssertions.
-- в тестовом классе Burger замокал Bun и Ingredient
-- в тесте testGetReceipt() в классе BurgerTest доработал проверку,теперь фактический вызов метода getReceipt() совпадает с ожидаемым результатом.
-- тест ingredientTypeTest() разбил на два.

EgorKoshelev96 committed:
Merge remote-tracking branch 'origin/develop1' into develop1

alesyaisachenko commented on pom.xml:
```diff
@@ -5,12 +5,136 @@
+        </dependency>
+        <dependency>
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
```
Это нужно доработать:

в проекте может быть или testNg или JUnit, в одном проекте две эти библиотеки не могут быть. Это библиотеки -- тестовые фреймворки. Они используются для написания тестов -- какие методы это тесты, как эти тесты запускать и так далее.

бибилиотеку testNg  из pom-файла нужно удалить

alesyaisachenko commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,111 @@
+                1000.0F
+        );
+        String actualReceipt = burger.getReceipt();
+        assertEquals(expectedReceipt, actualReceipt, String.format("Некорректный чек.Ожидалось: " + expectedReceipt + "Пришло: " + actualReceipt));
```
Можно лучше:
Со String.format работают так:

String.format("Некорректный чек.Ожидалось: %s, Пришло: %s",expectedReceipt, actualReceipt)

EgorKoshelev96 committed:
-- убрал библиотеку testNg переделал тесте по JUnit

-- оправил тестовый метод testGetReceipt(),а именно проваерку где используется String.format
